web/src/api/images/delete_all_images.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as getUnverifiedImagesSDK } from "@/api/images/graphql/getUnverifiedImages.generated";
+import { Session, getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import {
+  S3Client,
+  ListObjectsCommand,
+  DeleteObjectCommand,
+} from "@aws-sdk/client-s3";
+import { logger } from "@/lib/logger";
+
+export type DeleteAllImagesResponse = {
+  success?: boolean;
+  message?: string;
+};
+
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+});
+
+export type DeleteAllImagesBody = yup.InferType<typeof schema>;
+// This endpoint takes in an App ID and deletes all the images in the folder. Used if the user deletes the app.
```

Comment:
1. how does a user then delete a single image? 
2. this should be an internal call, triggered from a delete hook from Hasura. also just in case, might be best to expire the images and let's add a lifecycle rule to the bucket (infra) to delete all expired objects after 2-3 days.

Reply:
I was mainly using this to clean up old resources if the user deleted the app. For verified images they will overwrite the old image to replace it and can't just delete an image. For unverified, the bucket path is protected and the hasura is updated to not have an img_url

Reply:
Figure not clearing the image would accommodate for the use case where they click the x on the image but then don't save the changes

Reply:
Aligned in slack pushing this to another version